File: ios/Runner/Database/Models/Group/GroupInnerConversationModel.swift
//
//  GroupInnerConversationModel.swift
//  Runner
//
//  Created by Hashim Khan on 16/12/2024.
//


import Foundation

struct GroupInnerConversationModel: Codable, Equatable {
    var id: Int?
    var groupName: String?
    var modelId: Int?
    var modelType: ModelType?
    var name: String?
    var type: String?
    var image: String?
    var participants: [ParticipantModel]?
    var chatAble: Bool?
    var status: String?
    var dateTimeInHumans: String?
    var lastMessagedAt: Int?
    var unreadCount: Int?
    var mentioned: [Int]?
    var notificationMuted: Bool?

    enum CodingKeys: String, CodingKey {
        case id
        case groupName = "group_name"
        case modelId = "model_id"
        case modelType = "model_type"
        case name
        case type
        case image
        case participants
        case chatAble = "chat_able"
        case status
        case dateTimeInHumans = "date_time_in_humans"
        case lastMessagedAt = "last_messaged_at"
        case unreadCount = "unread_count"
        case mentioned
        case notificationMuted = "notification_muted"
    }
}

File: android/app/src/main/kotlin/com/transport_system/ts_admin/data_providers/models/group/GroupInnerConversationModel.kt
package com.transport_system.ts_admin.data_providers.models.group

import com.google.gson.annotations.SerializedName
import com.transport_system.ts_admin.data_providers.models.common.ModelType
import com.transport_system.ts_admin.data_providers.models.common.ParticipantModel

data class GroupInnerConversationModel(
    val id: Int?,
    @SerializedName("group_name") val groupName: String?,
    @SerializedName("model_id") val modelId: Int?,
    @SerializedName("model_type") val modelType: ModelType?,
    val name: String?,
    val type: String?,
    val image: String?,
    val participants: List<ParticipantModel>?,
    @SerializedName("chat_able") val chatAble: Boolean?,
    val status: String?,
    @SerializedName("date_time_in_humans") val dateTimeInHumans: String?,
    @SerializedName("last_messaged_at") val lastMessagedAt: Int?,
    @SerializedName("unread_count") val unreadCount: Int?,
    val mentioned: List<Int>?,
    @SerializedName("notification_muted") val notificationMuted: Boolean?
)
